vision_datasets/commands/utils.py

```diff
@@ -1,17 +1,91 @@
 
-logger = logging.getLogger(__name__)
-logging.basicConfig(level=logging.INFO)
+    return logger
+
+
+logger = set_up_cmd_logger(__name__)
 
 TSV_FORMAT_LTRB = 'ltrb'
 TSV_FORMAT_LTWH_NORM = 'ltwh-normalized'
 
 
+class Base64Utils:
+    def b64_str_to_pil(img_b64_str: str):
+        assert img_b64_str
+
+        return PILImageLoader.load_from_stream(io.BytesIO(base64.b64decode(img_b64_str)))
+
+    def file_to_b64_str(filepath: pathlib.Path):
+        fr = FileReader()
+        with fr.open(filepath.as_posix(), "rb") as file_in:
+            return base64.b64encode(file_in.read()).decode('utf-8')
+
+    def b64_str_to_file(b64_str: str, file_name: Union[pathlib.Path, str]):
+        assert b64_str
+        assert file_name
+
+        with open(file_name, 'wb') as file_out:
+            file_out.write(base64.b64decode(b64_str))
+
+
+def add_args_to_locate_dataset(parser):
+    parser.add_argument('name', type=str, help='Dataset name.')
+    parser.add_argument('--reg_json', '-r', type=pathlib.Path, default=None, help='dataset registration json file path.', required=False)
+    parser.add_argument('--version', '-v', type=int, help='Dataset version.', default=None)
+    parser.add_argument('--usages', '-u', nargs='+', choices=[Usages.TRAIN_PURPOSE, Usages.VAL_PURPOSE, Usages.TEST_PURPOSE],
+                        help='Usage(s) to check.', default=[Usages.TRAIN_PURPOSE, Usages.VAL_PURPOSE, Usages.TEST_PURPOSE])
+
+    parser.add_argument('--coco_json', '-c', type=pathlib.Path, default=None, help='Single coco json file to check.', required=False)
+    parser.add_argument('--data_type', '-t', type=str, default=None, help='Type of data.', choices=DatasetTypes.VALID_TYPES, required=False)
+
+    parser.add_argument('--blob_container', '-k', type=str, help='Blob container (sas) url', required=False)
+    parser.add_argument('--local_dir', '-f', type=pathlib.Path, required=False, help='Check the dataset in this folder. Folder will be created if not exist and blob_container is provided.')
+
+
+def get_or_generate_data_reg_json_and_usages(args):
+    def _generate_reg_json(name, type, coco_path):
+        data_info = [
+            {
+                'name': name,
+                'version': 1,
+                'type': type,
+                'format': 'coco',
+                'root_folder': '',
+                'train': {
+                    'index_path': coco_path.name
+                }
+            }
+        ]
+
+        return json.dumps(data_info)
+
+    if args.reg_json:
+        usages = args.usages or [Usages.TRAIN_PURPOSE, Usages.VAL_PURPOSE, Usages.TEST_PURPOSE]
+        data_reg_json = args.reg_json.read_text()
+    else:
+        assert args.coco_json, '--coco_json not provided'
+        assert args.data_type, '--data_type not provided'
+        usages = [Usages.TRAIN_PURPOSE]
+        data_reg_json = _generate_reg_json(args.name, args.data_type, args.coco_json)
+
+    return data_reg_json, usages
```

Comment:
@shonohs this unifies the dataset/[coco+images] access of multiple util scripts: download, od_to_ic onvert, to_tsv,